Source/WebCore/workers/service/context/ServiceWorkerThread.cpp

```
@@ -191,7 +191,11 @@ void ServiceWorkerThread::queueTaskToPostMessage(MessageWithMessagePorts&& messa
         if (std::holds_alternative<ServiceWorkerClientData>(sourceData)) {
             RefPtr<ServiceWorkerClient> sourceClient = ServiceWorkerClient::create(serviceWorkerGlobalScope, WTFMove(std::get<ServiceWorkerClientData>(sourceData)));
 
-            RELEASE_ASSERT(!sourceClient->url().protocolIsInHTTPFamily() || !serviceWorkerGlobalScope->url().protocolIsInHTTPFamily() || protocolHostAndPortAreEqual(serviceWorkerGlobalScope->url(), sourceClient->url()));
+            if (sourceClient->url().protocolIsInHTTPFamily() && serviceWorkerGlobalScope->url().protocolIsInHTTPFamily() && !protocolHostAndPortAreEqual(serviceWorkerGlobalScope->url(), sourceClient->url())) {
+                RELEASE_LOG(ServiceWorker, "ServiceWorkerThread::queueTaskToPostMessage host and port mismatch");
```

Feels like this should be a RELEASE_LOG_FAULT at least. Also, while this will address this crash, there are 2 concerns with this:
1. We may now crash later on, on a different assertion of the same kind
2. We may end up with a broken site now that we're silently dropping messages with a service worker.

We may not have a better solution in the short term but I hope we can look at the code and maybe come up with a speculative fix at least. I understand this is not easy in @youennf 's absence.